controllers/data.controller.js

```diff
@@ -178,8 +201,8 @@ exports.updateTierAndScore = async (req, res) => {
         isAnswerCorrect,
         questionsAnsweredNb,
       );
-      if (response.status === 200) {
-        return res.json({ message: "Tier and score updated successfully" });
+      if (response.status.code === 200) {
+        return res.status(200).json(response);
```

Comment:
please use http-status package for checking status. you can find example in spai-backend code. so, if everything is okay, it will just send res. therefore you dont need to send status and code from helper functions.